### ISSUE TITLE
[Bug]: Connecting an application to git throws an error - Git command execution error: Git command execution error: Remote does not have refs/heads/master available for fetch.

### ISSUE THREAD

**Issue by btsgh** [appsmithorg/appsmith#31787]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Description

When I connect the attached application to git, I get an error. 

### Steps To Reproduce

https://github.com/appsmithorg/appsmith/assets/101863839/d3dee322-c19b-470f-a806-f146d099febf

[OldApp_DSTesting1.9.24 (9).json](https://github.com/appsmithorg/appsmith/files/14599578/OldApp_DSTesting1.9.24.9.json)

### Environment

Production

### Severity

High (Blocker to building or releasing)

### Version

Cloud

**Comment by Nikhil-Nandagopal** [comment]
Closing this as it does not have an appropriate label associated with a pod

**Comment by riteshkew** [comment]
RCA : Failure was due to Github rejecting the commits, due to which the subsequent actions were failing. This was because of a recent change in code acceptance criteria for Github which rejects any code that it thinks contains sensitive information or keys. We found out that a twilio key was used directly i the application, which was causing the failure. We need to put better error handling around this failure, so marking down the severity accordingly.
